pkg/startup/v7/startup.go

```diff
@@ -119,7 +119,7 @@ func (s *startup) writeFiles(role api.AgentPoolProfileRole, w writers.Writer, ho
 			filepath = strings.TrimPrefix(filepath, "common")
 
 			filesToWrite[filepath] = string(b)
-
+			fileKeys = append(fileKeys, filepath)
```

Comment:
are you sure we should be changing this? we never create old versions (except in CI), they are there for hash comparisons really so you are now changing the old hashes..